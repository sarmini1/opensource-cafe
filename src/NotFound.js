import React from "react";
import {
  Card,
  CardBody,
  CardTitle
} from "reactstrap";

/** NotFound
 * 
 * Props:
 * - none
 * 
 * State:
 * - none
 * 
 * App --> Routes --> NotFound
 */
function NotFound() {

  return (
    <div>
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            <h1>This page could not be found.</h1>
          </CardTitle>
          <h2>To return home, click <a href="/">here.</a></h2>
        </CardBody>
      </Card>
    </div>
  )

}

export default NotFound;