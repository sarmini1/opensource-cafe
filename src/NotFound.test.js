import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

import NotFound from "./NotFound";

it('mounts without crashing', function () {
  render(<MemoryRouter><NotFound /></MemoryRouter>);
});

it("matches snapshot", function () {
  const { container } = render(
    <MemoryRouter><NotFound /></MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});

it("displays NotFound component with appropriate HTML", function () {
  const { debug, container } = render(
    <MemoryRouter><NotFound /></MemoryRouter>
  );

  debug(container);
  const notFound = container.querySelector(".NotFound");
  expect(notFound).toContainHTML(
    '<h2>To return home, click <a href="/">here.</a></h2>'
  )
});