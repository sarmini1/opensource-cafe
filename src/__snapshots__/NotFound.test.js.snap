// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="NotFound"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="font-weight-bold text-center card-title"
        >
          <h1>
            This page could not be found.
          </h1>
        </div>
        <h2>
          To return home, click 
          <a
            href="/"
          >
            here.
          </a>
        </h2>
      </div>
    </div>
  </div>
</div>
`;
